src/compiler/llvm-to-backend/host/LLVMToHost.cpp

```diff
@@ -101,6 +102,7 @@ bool LLVMToHostTranslator::toBackendFlavor(llvm::Module &M, PassHandler &PH) {
   registerCBSPipeline(MPM, hipsycl::compiler::OptLevel::O3, true);
 
   llvm::FunctionPassManager FPM;
+  FPM.addPass(HostKnownWgSizePass{KnownGroupSizeX, KnownGroupSizeY, KnownGroupSizeZ});
```

Comment:
While in general, I'm in favor of splitting things in different passes, I'm not entirely sure, what the benefit is of doing this in a separate pass compared to doing it here:
https://github.com/AdaptiveCpp/AdaptiveCpp/blob/bf218d9b835efe2ac3c609a10abb1b997b00b116/src/compiler/llvm-to-backend/host/HostKernelWrapperPass.cpp#L177

One benefit of keeping it as-is, would be the following optimization: only add the pass if any of the `KnownGroupSize[X-Z]` is not `0`.
Did you observe behavior that would make it important to do it the way it currently is?